File: src/main/java/br/com/blz/testjava/domain/entity/Product.java
package br.com.blz.testjava.domain.entity;

import br.com.blz.testjava.domain.objectvalue.Inventory;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@EqualsAndHashCode
public class Product {

    private Long sku;
    private String name;
    private Inventory inventory;
    private boolean isMarketable;

    public Product(Long sku, String name, Inventory inventory) {
        this.sku = sku;
        this.name = name;
        this.inventory = inventory;
    }
}

File: src/main/kotlin/br/com/blz/testjava/domain/entity/Product.kt
package br.com.blz.testjava.domain.entity

data class Product(
    val sku: Int,
    val name: String,
    val inventory: Inventory,
) {

    var isMarketable: Boolean = false
      get() = inventory.quantity > 0

    fun equals(other: Product): Boolean {
        return this.sku == other.sku
    }

}
